tests/test_batch.py

```diff
@@ -0,0 +1,364 @@
+
+
+JOB_OUT_PATH = "_job_test.out"
+JOB_ERR_PATH = "_job_test.err"
+
+
+# noinspection PyPep8Naming
+class BatchJobTest(unittest.TestCase):
+    io_collector = IOCollector()
+
+    def setUp(self) -> None:
+        self.io_collector.reset_paths()
+        self.io_collector.add_path(JOB_OUT_PATH)
+        self.io_collector.add_path(JOB_ERR_PATH)
+
+    def tearDown(self) -> None:
+        self.io_collector.delete_paths()
+
+
+class DryRunJobTest(BatchJobTest):
+
+    def test_job_ok(self):
+        job = DryRunJob.submit_job(['nc2zarr', '--help'], JOB_OUT_PATH, JOB_ERR_PATH)
+        self.assertIsInstance(job, DryRunJob)
+        self.assertIs(job.status, JobStatus.COMPLETED)
+        self.assertTrue(job.done)
+
+
+class LocalJobTest(BatchJobTest):
+
+    def test_job_ok(self):
+        job = LocalJob.submit_job(['nc2zarr', '--help'],
+                                  JOB_OUT_PATH,
+                                  JOB_ERR_PATH,
+                                  env_vars=dict(TEST='42'),
+                                  cwd_path='.',
+                                  poll_period=0.1)
+        self.assertIsInstance(job, LocalJob)
+        self.assertFalse(job.done)
+        while job.status is JobStatus.RUNNING:
+            time.sleep(0.1)
+        self.assertIs(job.status, JobStatus.COMPLETED)
+        self.assertTrue(job.done)
+        state = job.state
+        self.assertIsInstance(state, dict)
+        self.assertIn('exit_code', state)
+        self.assertIn('pid', state)
+
+
+SBATCH_MOCK_WIN32 = 'sbatch-mock.bat'
+SBATCH_MOCK_UNIX = './sbatch-mock.sh'
+
+SQUEUE_MOCK_WIN32 = 'squeue-mock.bat'
+SQUEUE_MOCK_UNIX = './squeue-mock.sh'
+
+
+class SlurmJobTest(BatchJobTest):
+
+    def setUp(self) -> None:
+        super().setUp()
+        if sys.platform == 'win32':
+            self.sbatch_program = SBATCH_MOCK_WIN32
+            self.squeue_program = SQUEUE_MOCK_WIN32
+        else:
+            self.sbatch_program = SBATCH_MOCK_UNIX
+            self.squeue_program = SQUEUE_MOCK_UNIX
+        self.io_collector.add_path(self.sbatch_program)
+        self.io_collector.add_path(self.squeue_program)
+
+    def _write_sbatch_exe(self, unix_content: str, win32_content: str):
+        self._write_exe(self.sbatch_program, unix_content, win32_content)
+
+    def _write_squeue_exe(self, unix_content: str, win32_content: str):
+        self._write_exe(self.squeue_program, unix_content, win32_content)
+
+    @classmethod
+    def _write_exe(cls, path: str, unix_content: str, win32_content: str):
+        if sys.platform == 'win32':
+            with open(path, 'w') as fp:
+                fp.write(win32_content)
+        else:
+            with open(path, 'w') as fp:
+                fp.write(unix_content)
+            st = os.stat(path)
+            os.chmod(path, st.st_mode | stat.S_IEXEC)
+
+
+SBATCH_OUT = 'Submitted batch job 43862490'
+
+SQUEUE_OUT_OK_1 = '             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)'
```

Comment:
```suggestion
SQUEUE_OUT_OK_1 = r'             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST\(REASON\)'
```

Same as Tonio's suggestion, but with a raw string to avoid creating invalid escape sequences which may break in a future Python version (see https://bugs.python.org/issue27364).